### DISCUSSION TITLE
Netbox Docker migration questions

### DISCUSSION THREAD

**Discussion by EHRETic** [netbox-community/netbox-docker#1021] [Getting Help] [answered]
Hi everyone,

In a desire to move all my webapps to a docker platform, I'd also like to move Netbox, which currently runs in a dedicated RHEL VM (with DB & Redis locally).

However, I'm having a bit of trouble understanding all the details of the docker compose, especially for my environment where Docker runs on a macvlan network (each container has its own IP).

For the PostgreSQL part, no problem, I have a centralized server, so all I have to do is migrate my DB, set the IP of the new server and the credentials.

For Redis, not complicated either, but as I don't know much about it, do I really need 2 instances (redis & redis-cache)? Can you use a single instance shared with other containers?

After that, it's really the rest that I'm wondering about: I don't really understand how the netbox-worker and netbox-housekeeping apps work in relation to the main application. Are they additional containers (which may seem a silly question, but it's hard to find precise information on the "<<:" syntax)? Whether they're additional containers or not, how should I configure them in my macvlan environment?

I know, that's a lot of questions, but I can't wait to get started on this migration! Thanks in advance for your help 😉

**Comment by EHRETic** [comment]
Anyone? 😊

**Comment by kkthxbye-code** [comment, marked as answer]
> do I really need 2 instances (redis & redis-cache)? Can you use a single instance shared with other containers?

Yes, you can easily use the same redis server. There's no real reason to run two as the cache redis is barely used.

>  I don't really understand how the netbox-worker and netbox-housekeeping apps work in relation to the main application. Are they additional containers (which may seem a silly question, but it's hard to find precise information on the "<<:" syntax)? Whether they're additional containers or not, how should I configure them in my macvlan environment?

Yes, they are additional containers. The worker container runs an `rqworker` that executes custom scripts, reports and webhooks. It's also used for the data sync feature. The housekeeping container just runs the housekeeping command which removes stuff that's older then the set retention limit. I don't know what macvlan is so can't help you there.

**Comment by EHRETic** [comment]
Hi 

Forgot to say thanks and to mention that I did migrate my instance successfully!👌

I just regret my topology map plugin but I think this should be a feature request to make the plugin installation easier so docker users can benefit from them in a more convenient way... 

Cheers 😉
